.github/workflows/linter.yml

```diff
@@ -0,0 +1,32 @@
+# Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+# See the file LICENSE for licensing terms.
+
+name: Run Golang Linter
+
+on:
+  push:
+    branches:
+      - '*'
```

Comment:
I think we should run this in the same way as most of our other checks - on all PRs and also pushes to main.

Reply:
oh, does this not check all PRs and push to main? The [test.yml](https://github.com/ava-labs/awm-relayer/blob/main/.github/workflows/test.yml#L6-L9) did the same.

Reply:
Discussed in the standup yesterday. The lint is light weighted. It should should be fine running for every push.